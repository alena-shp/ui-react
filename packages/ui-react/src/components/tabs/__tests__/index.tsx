import React from 'react';
import { create } from 'react-test-renderer';

import { TabsComposition } from '..';

test('Default use', () => {
    const tree = create(
        <TabsComposition
            items={[{
                id: '1',
                elTab: 'Tab #1',
                elPanel: 'Panel #1'
            }]}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Without ID', () => {
    const tree = create(
        <TabsComposition
            items={[{
                elTab: 'Tab #1',
                elPanel: 'Panel #1'
            }]}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Empty', () => {
    const tree = create(
        <TabsComposition />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Function elements', () => {
    const tree = create(
        <TabsComposition
            items={[{
                elTab: () => 'Tab #1',
                elPanel: () => 'Panel #1'
            }]}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});
