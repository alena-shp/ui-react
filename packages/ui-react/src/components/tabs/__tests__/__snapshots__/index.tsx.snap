// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default use 1`] = `
<div
  className="tabs"
  data-rttabs={true}
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <ul
    className="tab-list"
    role="tablist"
  >
    <li
      aria-controls="panel:r0:0"
      aria-disabled="false"
      aria-selected="true"
      className="tab tab_preset-pane tab_selected"
      data-rttab={true}
      id="tab:r0:0"
      role="tab"
      tabIndex="0"
    >
      Tab #1
    </li>
  </ul>
  <div
    aria-labelledby="tab:r0:0"
    className="tab-panel tab-panel_preset-pane tab-panel_selected"
    id="panel:r0:0"
    role="tabpanel"
  >
    Panel #1
  </div>
</div>
`;

exports[`Empty 1`] = `null`;

exports[`Function elements 1`] = `
<div
  className="tabs"
  data-rttabs={true}
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <ul
    className="tab-list"
    role="tablist"
  >
    <li
      aria-controls="panel:r2:0"
      aria-disabled="false"
      aria-selected="true"
      className="tab tab_preset-pane tab_selected"
      data-rttab={true}
      id="tab:r2:0"
      role="tab"
      tabIndex="0"
    >
      Tab #1
    </li>
  </ul>
  <div
    aria-labelledby="tab:r2:0"
    className="tab-panel tab-panel_preset-pane tab-panel_selected"
    id="panel:r2:0"
    role="tabpanel"
  >
    Panel #1
  </div>
</div>
`;

exports[`Without ID 1`] = `
<div
  className="tabs"
  data-rttabs={true}
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <ul
    className="tab-list"
    role="tablist"
  >
    <li
      aria-controls="panel:r1:0"
      aria-disabled="false"
      aria-selected="true"
      className="tab tab_preset-pane tab_selected"
      data-rttab={true}
      id="tab:r1:0"
      role="tab"
      tabIndex="0"
    >
      Tab #1
    </li>
  </ul>
  <div
    aria-labelledby="tab:r1:0"
    className="tab-panel tab-panel_preset-pane tab-panel_selected"
    id="panel:r1:0"
    role="tabpanel"
  >
    Panel #1
  </div>
</div>
`;
